Add a button binding to start the screensaver immediately

osaarima opened the pull request:
I added a button bind to start the screensaver immediately to fit it for my needs. The binding will also activate fullscreen and pause the video. Added instructions for the readme to reflect the changes.

osaarima committed:
Added a key binding to start screensaver immediately

osaarima committed:
Update README.md about button binding

Instructions for how to rebind the button.

Akemi commented:
just a heads up, will try to have a look at this, this weekend hopefully.

Akemi commented on oled-screensaver.lua:
```diff
@@ -155,3 +163,5 @@ state.mouseMovementTimer:kill()
 mp.observe_property("fullscreen", "bool", fullscreenEvent)
 mp.observe_property("seeking", "bool", clearEvent)
+
+mp.add_key_binding("Ctrl+p", "pause_and_start_screensaver", pauseAndStartScreensaver)
```
i would prefer not to bind this to any key by default (to not collide with other bindings in the future). readme should just explicitly mention that it needs to be bound manually.

also newline at the end of the file.

Akemi commented on oled-screensaver.lua:
```diff
@@ -145,6 +145,14 @@ function updateState()
     end
 end
 
+function pauseAndStartScreensaver()
```
can we add two different function/bindings here? that way the user can decide which behaviour he wants.
- one that automatically sets pause+fs
- one that only starts the screensaver when already paused+fs

Akemi commented on README.md:
````diff
@@ -1,7 +1,12 @@
+fullscreen. This can be rebinded in input.conf with
+```
+<new bind> script-binding pause_and_start_screensaver
+```
````
lets add a new paragraph for the binding docs. also adjust the text a bit pls. i leave the naming of the script bindings to your discretion.

something like:
> The screensaver can be manually started by binding a key to the appropriate predefined script bindings `start_screensaver` or `pause_and_start_screensaver`. while former only starts the screensaver when already paused and in fullscreen, the latter automatically and additionally sets mpv to pause and fullscreen. Both script bindings can be bound in your `input.conf` file with:
> ```
> < key> script-binding pause_and_start_screensaver
> ```

osaarima commented:
Thanks for the comments! I think each of these are good additions. I will do the modifications soon. I also noticed a bug which in some cases the screensaver does not start if the video is not in fullscreen and not paused. I think it might be that either the `observe_property` for pause and fullscreen or the mouse movevent causes the screensaver to end immediately after fullscreening. Nevertheless, I will also take a look at this.

osaarima committed:
Unbinded the default binding, added a new button binding, a bugfix

The bindings are now unbinded by default. A new button binding was added so that the user can choose the desired functionality when binding a button to this script. Also a bugfix was made for some cases when pauseAndStartScreensaver was used and the screensaver did not start automatically. Readme was updated.

osaarima committed:
Added a setting to select whether the mouse movement disables the screensaver

osaarima committed:
Fixed the default setting for mouse movement clearing to true

osaarima committed:
Changed ignore_clear_once to ignoreClearOnce to respect the camel case convention

osaarima commented:
I have now made the following fixes and additions:

- Default button bindings are removed.
- A new function bind has been added for activating the screensaver only when paused on fullscreen.
- README was updated with the comments in mind.
- A bugfix was made for cases when user activates the `pause_and_start_screensaver` binding but the screensaver does not activate. What sometimes happened was that the when the fullscreen was enabled the mouse movement binding for clearing the screensaver was also activated, thus ending the screensaver immediately. To fix this the mouse movement binding is ignored for one time and the problem goes away. This was done with a local state ignoreClearOnce.
- A new option mouseMovementClears was added. With this user can choose whether mouse movement will clear the screensaver or not. I have personally noticed that in some cases this is undesireable and would be nice to have an option to turn it off. If this is false the screensaver is cleared only by toggling pause or fullscreen, or by seeking. Default is set to true to respect the original behaviour. This has been updated to the README.
